x/auth/vesting/types/vesting_account.go

```diff
@@ -892,13 +895,15 @@ func (va *ClawbackVestingAccount) clawback(ctx sdk.Context, dest sdk.AccAddress,
 	}
 	addr := va.GetAddress()
 
+	// update the account's vesting settings
+	ak.SetAccount(ctx, va)
```

Comment:
Might want to note that this write is necessary in order to be able to transfer the unvested coins.